Fix for Multiple instances of node-java based programs

crimsonhawk009 opened the pull request.

jakobrun commented:
Thanks for the contribution. I wonder if we can add this feature without a breaking change on the api. For example by having a java option in the pool, connect and useInMemoryDb functions like this:
```javascript
const java = require('java')
const { pool } = require('node-jt400')
const connection = pool({java })
```
or we can also keep the pool, connect useInMemoryDb functions on top level implemented like this:
```javascript
const pool = options => initialize().pool(options)
```
What do you think @crimsonhawk009 ?

crimsonhawk009 commented:
@jakobrun 

Changed the pool, connect, and useInMemoryDb functions to take an optional java parameter and moved them back to the top level. The order of things need to be called changed. So I updated the README to follow.

crimsonhawk009 commented:
@jakobrun 
I have been running this build internally since last commit and has been working great.
Any chance of merging it?

jakobrun commented:
@crimsonhawk009 did you see my comment on the code? Also there is a conflict that needs to be resolved. Could you perhaps look at that?

crimsonhawk009 commented on lib/jt400.ts:
```diff
@@ -384,19 +365,65 @@ export interface InMemoryConnection extends Connection {
+}
+
+export function pool(config?, java?: nodeJava): Connection {
+	const jvm = initialize(java);
```
I thought about doing that but I only call pool once and export it. So I thought it would be an unneeded extra step.

Here is a snippet of what I do typically in my projects.

#### javaInit.js
```javascript
'use strict';

const java = require('java');
const jt400 = require('node-jt400');
const config = require('./config');

// Any custom java programs get put under java directory
// Any and all have to be initilized here
java.classpath.push(`${__dirname}/java`);
const iseries = jt400.pool(config, java);

// Custom node java applications
const customApp = java.import('com.testing.custom.app');

module.exports = {
        customApp,
	iseries,
};
```
I then run the functions on the exported pool.
```javascript
'use strict';
const { iseries } = require('./javaInit');

const getData = () => {
    try {
        const sql = 'select current date from sysibm.sysdummy1';
        const data = await iseries.query(sql);
        return data;
    }
    catch (err) {
        throw err;
    }
};
```

jakobrun replied:
Yes, but that is just one use case. It is possible to call pool multiple times, for example if you need connection to more than one system, and then java will be initialised each time.

crimsonhawk009 replied:
Makes sense. See latest commit.
